### PR TITLE
feat: multiple statuses reviewable

### PR DESCRIPTION
**Pull request by Zacharis278**
### [MST-2701](https://2u-internal.atlassian.net/browse/MST-2071)

- also added link to the proctoring tool dash, that got missed from the original story adding this modal.

code is still a work in progress but would like to get a review on the UX direction.

Screenshot 2023-10-13 at 10 48 35 AM

Screenshot 2023-10-13 at 10 27 34 AM

Screenshot 2023-10-13 at 11 02 10 AM

Screenshot 2023-10-13 at 10 48 20 AM

### CODE REVIEW

**Comment by MichaelRoytman** [comment]
These are my initial thoughts. I know some of these touch comments on previous UX/UI decisions, so feel free to disregard those, but they come to mind in light of this modal being used for additional exam attempt states.

1. The different modals for the three different exam attempt states are very similar visually. My concern is that we're not highlighting the different states prominently enough, and instructors may not realize they're looking at an exam attempt with a verified review instead of one that requires a secondary review if they're not careful and could make an inappropriate choice. Is there a way we can make that clearer within the modal? For example, in the title? Bolded text in the copy? Or maybe we could echo the color in the "Review" column button in the modal somehow (e.g. red for second review required and error and blue for verified)?

2. I also feel like the title "Please confirm your choice" doesn't make sense because the choice is being made in the modal itself. It feels like a premature confirmation modal. I wonder if the title should, instead, reiterate the state of the exam (e.g. "Manage verified attempt") or have more state relevant language (e.g. "Review attempt" or "Override attempt review"). That will also help to reiterate the difference in the modals.

3. As a follow up to #2, if we want a confirmation step, should the selection of "verify" or "reject" _then_ open a confirmation modal that actually confirms the choice? I don't want to necessarily require a ton of button clicks, but it could make more sense as a confirmation step.

5. I think that the amount of buttons and actions is a little overwhelming. What do you think about making `Review Dashboard` a link that opens in a new tab? We could either hyperlink the actual `review dashboard` copy in the paragraph or just turn the `Review Dashboard` button into a link. I think that would make it feel more like a secondary or supplementary action instead of a primary action like verify/reject.

6. We could differentiate between the "verify" and "reject" buttons through color. Styling "reject" as red could also help to differentiate between the two actions and better indicate what the buttons do.

7. In the "No (Cancel)" button, "No" feels like an odd word to use because there's no question being asked. Does "Cancel" suffice?

8. I think when an exam attempt is in the "second review required" state, the "verify" and "reject" actions are a little more straightforward. But for the "error" or "verified" states, especially, I think the impact of "verify" and "reject" are less clear. What does marking a "verified" attempt "verified" mean? Should that even be a valid option? What does it mean to "reject" an "errored" attempt? I think either describing the impact of the actions in the modal copy or using a tooltip to explain what the buttons do could be helpful.

**Comment by Zacharis278** [comment]
@MichaelRoytman shoot missed your comment and kept going on this. Taking a look at that feedback now

1. I'd agree the modal can be more clear the attempt already has a review status. I can make that stand out a bit more and to your other point, the title of this modal can also change to help that out. I do like the idea of having these statuses be colored icons but I don't want to go off adding things until we get some feedback.
2. I'll update this title
3. I don't think we need to add a second confirmation model. Seems excessive.
4. The review dashboard does open a new tab, I can try out some different styling so it appears less like the verify/reject buttons. I have it set as the default styling for a 'secondary' button right now maybe the inverted/outline secondary would work better.
5. Don't see any issue with that.
6. Cancel seems better to me.
7. I think this might be a case where this modal should have help text directing users to the official documentation on how to manage exam attempts. I think explaining how everything works here would be a bit verbose even today we rely on course teams to know what reset/resume/etc do. Oh, but it would be easy to disable making a no-op change (verified->verified, rejected->rejected) so I can add that.

also cc @ilee2u, some of these changes are purely alterations to the existing design. I think you worked with Gabe on this so let us know if there are reasons not to make alterations.

**Comment by Zacharis278** [comment]
updated UI

Screenshot 2023-10-16 at 2 55 49 PM

File: src/pages/ExamsPage/components/ReviewExamAttemptButton.test.jsx [hunk 1]
```diff
@@ -30,34 +31,65 @@ describe('ReviewExamAttemptButton', () => {
...
   });
   it('Modal appears upon clicking button', () => {
-    render(reviewButton);
-    screen.getByText('Second Review Required').click();
-    expect(screen.getByText('Please confirm your choice.')).toBeInTheDocument();
+    render(reviewButton());
+    screen.getByText('Review Required').click();
+    expect(screen.getByText('Update review status')).toBeInTheDocument();
   });
   it('Clicking the No button closes the modal', () => {
```
Comment (MichaelRoytman) [inline src/pages/ExamsPage/components/ReviewExamAttemptButton.test.jsx:41]:
```suggestion
  it('Clicking the Cancel button closes the modal', () => {
```

**Review by MichaelRoytman — APPROVED** [review]
This looks good! This is a much better experience than the legacy dashboard 🎉. I left a few nits.

File: src/pages/ExamsPage/components/ReviewExamAttemptButton.jsx [hunk 2]
```diff
@@ -55,18 +108,20 @@ const ReviewExamAttemptButton = ({
               {formatMessage(messages.ReviewExamAttemptButtonCancel)}
             </ModalDialog.CloseButton>
             <Button
-              variant="primary"
+              variant="success"
               onClick={e => { // eslint-disable-line no-unused-vars
                 modifyExamAttempt(attemptId, constants.ExamAttemptActions.verify);
               }}
+              disabled={attemptStatus === constants.ExamAttemptStatus.verified}
```
[the remark this answers is not in the document]
Comment (Zacharis278) [reply src/pages/ExamsPage/components/ReviewExamAttemptButton.jsx:115]: possibly, our disabled styling is kinda non-obvious in this case. I'll try out removing it and see how that looks.

**Pull request merged**